### ISSUE TITLE
He-rich stars being directed to step_HMS_HMS_RLO

### ISSUE THREAD

**Issue by celiotine** [POSYDON-code/POSYDON-public#448]
In the initial_eccentricity_flow_chart, both H-rich and He-rich stars in RLO are being directed to step_HMS_HMS_RLO. However, this step does not actually handle the He-rich stars, it just fails them because they are not in the `FOR_RLO_STATES`. This should be corrected so that these stars are better handled in the flow, ensuring the changes are also consistent with the error catching for RLO binaries with H-rich stars in the detached step (e.g. PR #430).

**Comment by mkruckow** [comment]
I agree, that we should handle them better. I think the reason why He-rich stars got sent to HMS-HMS_RLO is the fact, that at high metallicity we might find He-rich stars in the HMS-HMS_RLO gird. Ideally, we should have separate grids involving He-rich stars.
So, the change to/I'll do is: to use the same lists in the flow and the step. The question, what should be in or outside the list is the more complicated part.
